docs/beyond_basics/101-164-dataladdening.rst

```diff
@@ -0,0 +1,141 @@
+You should consider the size, file type and modification frequency of files in your decisions as well as potential plans to share a dataset with a particular third party infrastructure.
+
+Step 2: Dataset creation
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Transforming a directory into a dataset is done with :command:`datalad create --force`.
+The ``-f``/``--force`` option enforces dataset creation in non-empty directories.
+Consider :ref:`applying procedures <procedures>` with ``-c <procedure-name>`` do apply configurations that suit your use case.
+
+.. findoutmore:: What is my directory is already a Git repository?
+
+   If you want to transform a Git repository to a DataLad dataset, a :command:`datalad create -f` is the way to go, too, and completely safe.
+   Your Git history will stay intact and will not be tempered with.
+
+If you want to transform a series of nested directories into nested datasets, continue with :command:`datalad create -f` commands in all further subdirectories.
+
+.. findoutmore:: One or many datasets?
+
+   In deciding how many datasets you need, try to follow the benchmarks in chapter :ref:`chapter_gobig` and the yoda principles in section :ref:`yoda`.
+   Two simple questions can help you make a decision:
+
+   #. Do you have independently reusable components in your directory, for example data from several studies, or data and code/results? If yes, make each individual component a dataset.
+   #. How large is each individual component? If it exceeds 100-200k files, split it up into smaller datasets. The decision on where to place subdataset boundaries can be guided by the existing directory structure or by common access patterns, for example based on data type (raw, processed, ...) or subject association. One straightforward organization may be a top-level superdataset and subject-specific subdatasets, mimicking the structure chosen in the use case :ref:`usecase_HCP_dataset`.
+
+You can automate this with :term:`bash` loops, if you want.
+
+.. findoutmore:: Example bash loops
+
+   Consider a directory structure that follows a naming standard such as `BIDS <https://bids.neuroimaging.io/>`_::
+
+      # create a mock-directory structure:
+      $ mkdir -p study/sub-0{1,2,3,4,5}/{anat,func}
+      $ tree study
+      study
+        ├── sub-01
+        │   ├── anat
+        │   └── func
+        ├── sub-02
+        │   ├── anat
+        │   └── func
+        ├── sub-03
+        │   ├── anat
+        │   └── func
+        ├── sub-04
+        │   ├── anat
+        │   └── func
+        └── sub-05
+            ├── anat
+            └── func
+
+   Consider further that you want to transform all ``sub-*`` directories into individual datasets.
+   Here is a line that would do this for the example above::
+
+      $ for dir in study/sub-0{1,2,3,4,5}; do datalad -C $dir create --force; done
```

Comment:
-> `create -d^` (see above)

Reply:
mhhh, what am I doing wrong? This does not lead to individual subdatasets:

```sh
adina@muninn:/tmp$ mkdir -p top/sub-0{1,2,3,4,5}/{anat,func}
adina@muninn:/tmp$ echo 123456 | tee -a top/file_l1 top/sub-0{1,2,3,4,5}/file_l2123456
adina@muninn:/tmp$ tree top
top
├── file_l1
├── sub-01
│   ├── anat
│   ├── file_l2
│   └── func
├── sub-02
│   ├── anat
│   ├── file_l2
│   └── func
├── sub-03
│   ├── anat
│   ├── file_l2
│   └── func
├── sub-04
│   ├── anat
│   ├── file_l2
│   └── func
└── sub-05
    ├── anat
    ├── file_l2
    └── func

15 directories, 6 files
adina@muninn:/tmp$ datalad -C top create -f
[INFO   ] Creating a new annex repo at /tmp/top 
create(ok): /tmp/top (dataset)
adina@muninn:/tmp$ for dir in top/sub-0{1,2,3,4,5}; do datalad -C $dir create -d^ --force; done
[INFO   ] Creating a new annex repo at /tmp/top 
create(ok):sub-01/ /tmp/top (dataset)
[INFO   ] Creating a new annex repo at /tmp/top 
create(ok):sub-02/ /tmp/top (dataset)
[INFO   ] Creating a new annex repo at /tmp/top 
create(ok):sub-03/ /tmp/top (dataset)
[INFO   ] Creating a new annex repo at /tmp/top 
create(ok):sub-04/ /tmp/top (dataset)
[INFO   ] Creating a new annex repo at /tmp/top 
create(ok):sub-05/ /tmp/top (dataset)
adina@muninn:/tmp$ datalad -C top/ save -d . -r
add(ok): /tmp/top/file_l1 (file)                                                
add(ok): /tmp/top/sub-01/file_l2 (file)
add(ok): /tmp/top/sub-02/file_l2 (file)
add(ok): /tmp/top/sub-03/file_l2 (file)
add(ok): /tmp/top/sub-04/file_l2 (file)
add(ok): /tmp/top/sub-05/file_l2 (file)
save(ok): /tmp/top (dataset)
action summary:
  add (ok: 6)
  save (ok: 1)adina@muninn:/tmp$ tree top
top
├── file_l1 -> .git/annex/objects/p3/6j/MD5E-s14--89ca78c08cc4562d00e2cb2a1d67fcac/MD5E-s14--89ca78c08cc4562d00e2cb2a1d67fcac
├── sub-01
│   ├── anat
│   ├── file_l2 -> ../.git/annex/objects/22/Gw/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af
│   └── func
├── sub-02
│   ├── anat
│   ├── file_l2 -> ../.git/annex/objects/22/Gw/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af
│   └── func
├── sub-03
│   ├── anat
│   ├── file_l2 -> ../.git/annex/objects/22/Gw/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af
│   └── func
├── sub-04
│   ├── anat
│   ├── file_l2 -> ../.git/annex/objects/22/Gw/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af
│   └── func
└── sub-05
    ├── anat
    ├── file_l2 -> ../.git/annex/objects/22/Gw/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af/MD5E-s7--f447b20a7fcbf53a5d5be013ea0b15af
    └── func

15 directories, 6 files

adina@muninn:/tmp$ datalad -C top subdatasets
-> None
```